### PR TITLE
Test: Refine annotaion and testcase for poll.

### PR DESCRIPTION
**Pull request by youhuazhu**
1. Add more detail info to make the purpose and process
of the test cases more clear which include test goal,
test step, input, judging criteria, constraints, etc.,
and these can be seen in our Zephyr documentations.
2.  Add some negative test code.

Signed-off-by: YouhuaX Zhu youhuax.zhu@intel.com

### CODE REVIEW

File: tests/kernel/poll/src/test_poll.c [hunk 1]
```diff
@@ -253,13 +253,48 @@ void check_results(struct k_poll_event *events, uint32_t event_type,
...
+ * - CONFIG_DYNAMIC_OBJECTS
+ * - CONFIG_POLL
+ *
+ * Input Specifications:
+ * - N/A
+ *
+ * Test Procedure:
+ * -# Use FIFO/semaphore/signal object to define poll event.
+ * -# Initialize the FIFO/semaphore/signal object.
+ * -# Create a sub thread to put FIFO,
```
Comment (LeiW000) [inline tests/kernel/poll/src/test_poll.c:281]: I think thatn "sub" can be removed.

File: tests/kernel/poll/src/test_poll.c [hunk 2]
```diff
@@ -253,13 +253,48 @@ void check_results(struct k_poll_event *events, uint32_t event_type,
...
+ * give semaphore and raise signal.
+ * -# Check the result when signal be raised, semaphore be given
```
Comment (LeiW000) [inline tests/kernel/poll/src/test_poll.c:283]: It better to change it with:

"Check the result when signal is raised, semaphore is givn and fifo is filled."

Also, plese check line 285 ~ 288.

File: tests/kernel/poll/src/test_poll.c [hunk 3]
```diff
@@ -344,8 +382,10 @@ void test_poll_wait(void)
 	check_results(&wait_events[1], K_POLL_TYPE_DATA_AVAILABLE, false);
 	check_results(&wait_events[2], K_POLL_TYPE_SIGNAL, false);
 
-	/* fifo */
-	rc = k_poll(wait_events, ARRAY_SIZE(wait_events), K_SECONDS(1));
+	/* fifo, Time out can set to K_FOREVER since
+	 * main thread is high priority.
+	 */
+	rc = k_poll(wait_events, ARRAY_SIZE(wait_events), K_FOREVER);
```
Comment (LeiW000) [inline tests/kernel/poll/src/test_poll.c:388]: My understanding is that you think that "K_SECONDS(1)" should be replaced with "K_FOREVER".

**Review by LeiW000 — CHANGES_REQUESTED** [review]

File: tests/kernel/poll/src/test_poll.c (hunk 3 above)
Comment (pabigot) [reply tests/kernel/poll/src/test_poll.c:388]: I don't see a good reason to change this code.  What's the benefit of eliminating the timeout?

**Review by pabigot — COMMENTED** [review]
Not sure why I was asked to review, but one comment anyway.

File: tests/kernel/poll/src/test_poll.c (hunk 3 above)
Comment (youhuazhu) [reply tests/kernel/poll/src/test_poll.c]: As I re-check the test code,  It's better to align with context,  so  I revert my change in this place.

File: tests/kernel/poll/src/test_poll.c (hunk 2 above)
Comment (youhuazhu) [reply tests/kernel/poll/src/test_poll.c]: fixed,  thanks Steven.

File: tests/kernel/poll/src/test_poll.c (hunk 1 above)
Comment (youhuazhu) [reply tests/kernel/poll/src/test_poll.c]: fixed,  thanks Steven.

**Review by enjiamai — APPROVED** [review]
The overall content look ok for me, only a little thing need to notice . Capitalize the first word of the sentence. Not a big deal.

**Review by LeiW000 — APPROVED** [review]

**Comment by LeiW000** [comment]
@nashif , could you help to merge it?

**Review by nashif — APPROVED** [review]

**Pull request merged**